### PR TITLE
Add demand control for uplink

### PR DESCRIPTION
**Pull request by Y-Guo**
Implements https://github.com/mdg-private/planning/issues/727

This PR uses the new `minDelaySeconds` returned by uplink to set the polling interval instead of gateways polling every 10 seconds by default configurable on the gateway side. If no `minDelaySeconds` is provided by uplink, `pollIntervalInMs` will still be used.

### CODE REVIEW

**Comment by Y-Guo** [comment]
TODO write a test

**Comment by trevor-scheer** [comment]
Can we add a test that demonstrates gateway utilizing this value? Another test that exercises the Math.max logic (largest value overrides) would be good too.

I'm wondering if `pollIntervalInMs` should be renamed, this new behavior might be surprising just based on the name of the config option. In what case would there not be a designated poll interval coming from uplink? Can we just fall back to 10s in the case that it's not provided and remove the option from managed mode configurations altogether? Open to ideas here, I'm just throwing out the first things that come to mind.

File: gateway-js/src/supergraphManagers/UplinkFetcher/__tests__/loadSupergraphSdlFromStorage.test.ts [hunk 1]
```diff
@@ -388,10 +390,46 @@ describe("loadSupergraphSdlFromUplinks", () => {
...
       maxRetries: 5,
       roundRobinSeed: 0,
+      earliestFetchTime: null,
     });
 
     expect(result).toBeNull();
     expect(fetcher).toHaveBeenCalledTimes(1);
   });
+
+  it("Waits the correct time before retrying", async () => {
```
[the remark this answers is not in the document]
Comment (Y-Guo) [reply gateway-js/src/supergraphManagers/UplinkFetcher/__tests__/loadSupergraphSdlFromStorage.test.ts:400]: cc @trevor-scheer

**Review by trevor-scheer — COMMENTED** [review]
Thanks for the test addition 👍  The code changes LGTM.

I'm still concerned about `pollIntervalInMs`.

In managed mode, is there ever a case where the user's configuration is relevant after this change? Is it just the failure case?

I would be inclined to have a sane default which the gateway uses if it doesn't get one from Uplink and remove this notion of polling configurability altogether. Either that, or create a more appropriately named option for what this actually does now, which is more like `fallbackPollIntervalInMs`.

I think we should get some other opinions on this from @apollographql/atlas.

**Comment by Y-Guo** [comment]
From a quick look in the code it seems `pollIntervalInMs` is only used by the `UplinkFetcher`, `IntrospectAndCompose`, and `LegacyFetcher`. Assuming legacy doesn't matter, `IntrospectAndCompose` is only [initialized](https://github.com/apollographql/federation/blob/main/gateway-js/src/index.ts#L377) when `isServiceListConfig()` is true BUT there's also a check that [warns](https://github.com/apollographql/federation/blob/main/gateway-js/src/index.ts#L262) against using `pollIntervalInMs` and `isServiceListConfig()` at the same time!

**Comment by cpeacock** [comment]
I like the idea of setting it as `fallbackPollIntervalInMs` and creating less confusion with that value being far less useful as a result of this PR.  I think you're right that legacy doesn't matter in this case because it's not using uplink.  A sane default in the case of an outage may lead to a traffic stampede when service returns, so, while it's very unlikely to happen in practice, is something at the back of my head.

**Review by trevor-scheer — APPROVED** [review]
LGTM!

**Pull request merged**